manifests/charts/base/templates/validatingwebhookconfiguration.yaml

```diff
@@ -1,3 +1,4 @@
+{{ if or (eq .Values.revision "") (eq .Values.defaultRevision true) }}
```

Comment:
Doesn't this mean ALL revisions will be validators?

I am not sure we actually came to a conclusion on this, but at one point we had discussed having defaultRevision=true just create an Istiod Service instead, and keep this part the same. That has some benefits: we don't need `revision` in base (which we try to keep minimal configuration for), and the default Revision can be changed by only controlling the `istio-discovery` chart rather than updating the base chart. It also allows other services which may not care about revisions to just access any Istiod (not sure how useful this actually is in practice though...)

The benefits of this approach is that its impossible to create a configuration only valid for one version (this can also be a con). Downsides are that our availability is tied to *all* revisions working, extra latency

Reply:
My understanding is that this means "if there is no revision specified OR the user specified that this is the default revision" then overwrite the EXISTING validatingwebhook. Only one revision can be a validator at any given time since the webhook just gets overwritten to point to a different service, right?

I had another implementation that worked how you're describing with an `istiod-default` service. I saw some potential drawbacks with that approach including:

1. Two almost-identical configurations for the services (`istiod` and `istiod-default`)
2. Repointing the service without making the webhook `failurePolicy: Ignore` could cause a short window of failing validation (I'm not sure there's a way to address this without a controller? Or maybe it's not an issue?)

Reply:
Oh I read it wrong, thanks.

1. install with `--revision=foo --defaultRevision`. We create istiod-foo.
2. install with `--revision=bar --defaultRevision`. We create istiod-bar.

Now we have 2 validators. I don't think we would overwrite it, since the name is different?

Reply:
Ah I think I'm confused on what you mean by validators here. The *service* name is different, but the name of the `ValidatingWebhookConfiguration`, the thing we're overwriting, stays as `istiod-{{ .Values.global.istioNamespace }}`?

So you'd do `--revision=foo --defaultRevision` and have a single `ValidatingWebhook` called `istiod-istio-system` pointed to `istiod-foo`.

Then you'd do `--revision=bar --defaultRevision` and you'd still have a single `ValidatingWebhook` called `istiod-istio-system`, except now pointed to `istiod-bar`.

Reply:
I read it wrong :sleeping: